### PR TITLE
[14.0][ADD] account_invoice_payment_ref_mandatory

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Adds required payment_reference field

### CODE REVIEW

File: account_invoice_payment_ref_mandatory/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# © 2022 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+{
+    "name": "Account Invoice Payment Ref Mandatory",
+    "summary": """
+         Adds required payment_reference field
+    """,
+    "author": "Solvos",
+    "license": "AGPL-3",
```
Comment (adriresu) [inline account_invoice_payment_ref_mandatory/__manifest__.py]: All OK, but this LICENSE should be changed.
```suggestion
    "license": "LGPL-3",
```

**Review by adriresu — CHANGES_REQUESTED** [review]

File: account_invoice_payment_ref_mandatory/__manifest__.py (hunk 1 above)
Comment (ChristianSantamaria) [reply account_invoice_payment_ref_mandatory/__manifest__.py]: Done 👌
thanks for the review

File: account_invoice_payment_ref_mandatory/views/account_move_views.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+<odoo>
+
+    <record id="view_account_move_form_inherit" model="ir.ui.view">
+        <field name="name">account.move.form (Ref Mandatory)</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_move_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='payment_reference']" position="attributes">
+                <attribute name="required">1</attribute>
+            </xpath>
```
Comment (dalonsod) [inline account_invoice_payment_ref_mandatory/views/account_move_views.xml]: Payment reference is currently invisible for certain move types, see:

https://github.com/odoo/odoo/blob/14.0/addons/account/views/account_move_views.xml#L662-L663

Then, with this code, when the move is a journal entry, required and invisible should become incompatible (we should fill a field that is invisible).

Anyway, we'll only set required this field for Vendor operations (`in_invoice` and `in_refund`), we'll extended it later if needed.

Adding view inheritance please check if current `attrs` values (for `invisible` and `readonly`) are kept, or if we should need to copy/paste them in this view.

**Review by dalonsod — CHANGES_REQUESTED** [review]

**Review by dalonsod — APPROVED** [review]
We've got a corner case: a vendor draft invoice, maybe only when the invoice is published we should check this field as required, but this initial functionality is enough today (users should fill a dummy reference when the invoice is created).

Let's merge it...

**Pull request merged**